src/utils.js

```diff
@@ -42,6 +62,27 @@ function makeUrl() {
   });
 }
 
+/**
+ * Removes a prefix from the beginning of the string.
+ *
+ * @param str - The string to process.
+ * @param prefix - The prefix to remove.
+ * @param [limit] - Maximum amount of times to trim. No limit by default.
+ * @returns - The processed string.
+ */
+function trimPrefix(str, prefix, limit) {
+  while (str.startsWith(prefix)) {
+    if (limit !== undefined && limit <= 0) {
+      break;
+    }
+    str = str.slice(prefix.length);
+    if (limit) {
+      limit -= 1;
+    }
+  }
+  return str;
+}
+
```

Comment:
I think this might look a little more readable

```suggestion
function trimPrefix(str, prefix, limit) {
  const matcher = new RegExp(`^(${prefix}){0,${limit ?? ""}}`);
  return str.replace(matcher, "")
}

```

Reply:
I'm not too familiar with regexps and try not to use them. They tend to be hard to read and inefficient. I'm not sure that one will work if you substitute in `${prefix}` like that, and the second half with `limit` I unfortunately don't understand at all.

Reply:
I would also prefer the non-regexp version here. The regexp is shorter but it's a little harder to read and more prone to errors when modified. The while version is dead simple and immediately understandable.

Reply:
The original `trimPrefix` implementation is already flawed because it handles `null` and `undefined` differently. It's because you strict compare it to `undefined`, which is always a bit dangerous. 

The `RegExp` only looks complicated but it's honestly just a group and a quantifier. I do think the bullish coalescing here makes it hard to read. Extracting it into a variable and adding a comment would make it easy to read imo.

```
const repeats = `{0,${limit ?? ""}}` // trim all if limit is null or undefined
const matcher = new RegExp(`^(${prefix})${repeats}`);
return str.replace(matcher, "")
```

Reply:
Probably best to add a test case that covers both `null` and `undefined`.

Reply:
The function is taken from `skynet-js` where we have Typescript typing. The type of `limit` is `limit?: number`, so if someone were to try to pass `null` it wouldn't compile.

I don't think a regexp here is appropriate and I'm not sure it's performant to be compiling it every time the function is called. Let's not spend more resources on a regexp solution here, let's just keep what is already there and known to work.

Reply:
How can we approve this PR though. It adds a function which we know is flawed. If you pass `null` or `undefined` you won't get the same outcome. If you don't want to adopt the regex, why wouldn't we update the code to work properly.

Reply:
I added validation.